api/v1alpha1/workspace_types.go

```diff
@@ -73,6 +73,35 @@ type AccessStrategyRef struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
+// IdleShutdownSpec defines idle shutdown configuration
+type IdleShutdownSpec struct {
+	// Enabled indicates if idle shutdown is enabled
+	Enabled bool `json:"enabled"`
+
+	// TimeoutMinutes specifies idle timeout in minutes
+	// +kubebuilder:validation:Minimum=1
+	TimeoutMinutes int `json:"timeoutMinutes"`
+
+	// Detection specifies how to detect idle state
+	Detection IdleDetectionSpec `json:"detection"`
+}
+
+// IdleDetectionSpec defines idle detection methods
+type IdleDetectionSpec struct {
+	// EndpointCheck specifies HTTP endpoint to check for idle status
+	// +optional
+	EndpointCheck *EndpointCheckSpec `json:"endpointCheck,omitempty"`
+}
```

Comment:
non-blocking: should there be a `HealthCheckIntervalMinutes` attribute or similar?

Consider following the precedent of `ReadinessProbe`:
```bash
kubectl explain pod.spec.containers.readinessProbe
```